homeassistant/components/minecraft_server/__init__.py

```diff
@@ -219,49 +222,30 @@ async def _async_status_request(self) -> None:
 class MinecraftServerEntity(Entity):
     """Representation of a Minecraft Server base entity."""
 
+    _attr_should_poll = False
+
     def __init__(
-        self, server: MinecraftServer, type_name: str, icon: str, device_class: str
+        self,
+        server: MinecraftServer,
+        type_name: str,
+        icon: str,
+        device_class: str | None,
     ) -> None:
         """Initialize base entity."""
         self._server = server
-        self._name = f"{server.name} {type_name}"
-        self._icon = icon
-        self._unique_id = f"{self._server.unique_id}-{type_name}"
+        self._attr_name = f"{server.name} {type_name}"
```

Comment:
Maybe new entity naming style should be used here?
```python
        self._attr_name = type_name
```
and `_attr_has_entity_name = True`.

Reply:
I originally thought to keep the PR as small as possible, but I guess it makes sense to add here: 50ec45c1fda7ba92ba8ed71cf3df0802f5294101